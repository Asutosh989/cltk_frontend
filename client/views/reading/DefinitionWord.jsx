DefinitionWord = React.createClass({

  propTypes: {
    word: React.PropTypes.object.isRequired

  },

  render() {

     return (
          <div className="meta-item definition">
            <div className="show-more-toggle">
              <i className="mdi mdi-plus paper-shadow"></i>
              <i className="mdi mdi-minus paper-shadow"></i>

            </div>

            <div className="definition-title meta-item-title">
              <h4 className="word">
                {this.props.word.lemma}
              </h4>
              <a className="definition-link" href="http://www.perseus.tufts.edu/hopper/morph?l={this.props.word.lemma}&la=la" target="_blank">
                Perseus
              </a>
            </div>

            <div className="word-meanings">

              {this.props.word.definitions.map(function(definition){

                <div className="meaning">
                  <span className="root">{definition.orthography}</span>
                  <span className="meaning-definition">{definition.senses}</span>

                  {definition.inflections.map(function(inflection){
                    <div className="forms">

                      {(() => {
                        switch (inflection.pos) {
                          case "noun":
                            return <span className="form">-{inflection.ending}: {inflection.pos} {inflection.declension.title} declension, {inflection.declension.case} {inflection.gender} {inflection.number}</span>

                          case "pronoun":
                            return <span className="form">-{inflection.ending}: {inflection.pos} {inflection.declension.title} declension, {inflection.declension.case} {inflection.gender} {inflection.number}</span>

                          case "adjective":
                            return <span className="form">-{inflection.ending}: {inflection.pos} {inflection.declension.title} declension, {inflection.declension.case} {inflection.gender} {inflection.number}</span>

                          case "verb":
                            return <span className="form">-{inflection.ending}: {inflection.pos} {inflection.conjugation} conjugation, {inflection.person} {inflection.number} {inflection.tense} {inflection.voice} {inflection.mood}</span>

                          case "participle":
                            return <span className="form">-{inflection.ending}: {inflection.pos} {inflection.declension.title} declension, {inflection.declension.case} {inflection.gender} {inflection.number} {inflection.tense} {inflection.voice}</span>

                          default:
                            return <span className="form">-{inflection.ending}: {inflection.pos} {inflection.form}</span>

                          }

                      })()}

                    </div>

                  })}
                </div>

              })}

            </div>

          </div>

     );
   }

});
