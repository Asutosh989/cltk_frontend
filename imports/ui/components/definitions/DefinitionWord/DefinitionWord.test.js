
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import DefinitionWord from './DefinitionWord';

describe('DefinitionWord', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<DefinitionWord />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
