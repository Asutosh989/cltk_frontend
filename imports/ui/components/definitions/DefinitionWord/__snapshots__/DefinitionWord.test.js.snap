// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefinitionWord renders correctly 1`] = `
<div
  className="meta-item panel-item definition "
>
  <div
    className="show-more-toggle"
    onClick={[Function]}
  >
    <i
      className="mdi mdi-plus paper-shadow"
    />
    <i
      className="mdi mdi-minus paper-shadow"
    />
  </div>
  <div
    className="definition-title meta-item-title"
  >
    <h4
      className="word"
    />
    <a
      className="definition-link"
      href="http://www.perseus.tufts.edu/hopper/morph?l=undefined&la=la"
      rel="noopener noreferrer"
      target="_blank"
    >
      Perseus
    </a>
  </div>
  <div
    className="word-meanings"
  />
  <div
    className="bottom-gradient"
  />
</div>
`;
